src/bin/pgcopydb/ld_test_decoding.c

```diff
@@ -300,6 +302,55 @@ parseTestDecodingMessage(StreamContext *privateContext,
 }
 
 
+/*
+ * findIdentifierEndPos returns the first position of the separator in the given
+ * message after navigating through all the quotes. It assumes that the message
+ * contains a well-formed identifier and doesn't attempt to handle errors.
+ */
+static int
+findIdentifierEndPos(const char *message, char separator)
```

Comment:
I think it would be best if that handled being passed a NULL message, which would hint towards our usual approach of returning a boolean (success). Also, I suppose we could return the position where the dot is and the position of the first character after the identifier (or NULL), so we could probably use a `typedef struct` as a (modifiable) argument to the function.